torch/distributed/elastic/agent/server/api.py

```diff
@@ -557,12 +557,20 @@ def _rendezvous(self, worker_group: WorkerGroup) -> None:
         worker_group.group_world_size = group_world_size
 
         if group_rank == 0:
-            self._set_master_addr_port(
-                store,
-                spec.master_addr,
-                spec.master_port,
-                spec.local_addr,
-            )
+            # pyre-ignore [16]: Undefined attribute: torch._C._distributed_c10d.Store` has no attribute `check
+            if (
+                os.getenv("TORCH_SHARE_RDZV_TCP_STORE", "0") == "1"
```

Comment:
I'd prefer if this was an implementation detail within the rendezvous handler -- i.e. `if rdzv_handler.get_master_addr()`.